apps/emqx_rule_engine/src/emqx_rule_engine_api.erl

```diff
@@ -363,15 +395,36 @@ param_path_id() ->
             }),
             {500, #{code => 'INTERNAL_ERROR', message => ?ERR_BADARGS(Reason)}}
     end.
-'/rules/:id/reset_metrics'(put, #{bindings := #{id := RuleId}}) ->
-    case emqx_rule_engine_proto_v1:reset_metrics(RuleId) of
-        ok ->
-            {200, <<"Reset Success">>};
-        Failed ->
-            {400, #{
-                code => 'BAD_REQUEST',
-                message => err_msg(Failed)
-            }}
+
+'/rules/:id/metrics'(get, #{bindings := #{id := Id}}) ->
+    case emqx_rule_engine:get_rule(Id) of
+        {ok, _Rule} ->
+            NodeMetrics = get_rule_metrics(Id),
+            MetricsResp =
+                #{
+                    id => Id,
+                    metrics => aggregate_metrics(NodeMetrics),
+                    node_metrics => NodeMetrics
+                },
+            {200, MetricsResp};
+        not_found ->
+            {404, #{code => 'NOT_FOUND', message => <<"Rule Id Not Found">>}}
+    end.
+
+'/rules/:id/metrics/reset'(put, #{bindings := #{id := Id}}) ->
+    case emqx_rule_engine:get_rule(Id) of
+        {ok, _Rule} ->
+            case emqx_rule_engine_proto_v1:reset_metrics(Id) of
+                ok ->
+                    {204};
+                Failed ->
+                    {400, #{
```

Comment:
hard to reason why this should be a 400,
isn't this 500 ?

Reply:
You're right, doesn't make sense to be a 400. Also things get a lot easier by just letting it crash here.